code/modules/clothing/modular/armor_module.dm

```diff
@@ -0,0 +1,54 @@
+/**
+	Core armor module
+
+	This is the core of the armor modules
+	here we define the base attach / detach system of each module.
+*/
+/obj/item/armor_module
+	name = "armor module"
+	desc = "A dis-figured armor module, in its prime this would've been a key item in your modular armor... now its just trash."
+	icon = 'icons/mob/modular/modular_armor.dmi'
+
+	/// Is the module passive (always active) or a toggle
+	var/module_type = ARMOR_MODULE_PASSIVE
+
+
+/obj/item/armor_module/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+
+	if(!proximity_flag)
+		return
+	if(QDELETED(user) || user.stat != CONSCIOUS)
+		return
+	if(!istype(target, /obj/item/clothing/suit/modular))
+		return
+	var/obj/item/clothing/suit/modular/parent = target
+	if(!parent.can_attach(user, src))
+		return
+
+	do_attach(user, target)
```

Comment:
Why an `afterattack` here instead of an `attackby` on the `/obj/item/clothing/suit/modular` ?